instance/models/openedx_instance.py

```diff
@@ -127,14 +128,22 @@ def database_name(self):
         """
         The database name used for external databases/storages, if any.
         """
-        name = self.internal_lms_domain.replace('.', '_')
-        # Escape all non-ascii characters and truncate to 50 chars.
-        # The maximum length for the name of a MySQL database is 64 characters.
-        # But since we add suffixes to database_name to generate unique database names
-        # for different services (e.g. xqueue) we don't want to use the maximum length here.
-        allowed = string.ascii_letters + string.digits + '_'
-        escaped = ''.join(char for char in name if char in allowed)
-        return truncate_name(escaped, length=50)
+        current_database_name = self.current_database_name
+
+        if current_database_name is not None and current_database_name == '':
```

Comment:
While this would work, we use a similar pattern elsewhere in the code where we set the values inside the `save()` method, see [here](https://github.com/open-craft/opencraft/blob/8b9634ff96c97e48c384353a2baec0ffe90d08d4/instance/models/openedx_instance.py#L154-L161) for example.
This also lets us set the field to `blank=False` which ensures that no new instance is created with the particular field left empty.